agent/internal/agent.go

```diff
@@ -513,14 +515,19 @@ func (a *agent) postTaskLog(log model.TaskLog) error {
 	return nil
 }
 
-func runAPIServer(options Options, system *actor.System) error {
```

Comment:
This seems like a fairly major change, I wonder if we can avoid it? Maybe if we added `OtelEnabled` to `Options`, and switched it on within `Run`?

Reply:
Yes that's true. I'll look into passing it through options.